src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/producer/OutgoingMessageProducer.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.companieshouse.chipsrestinterfacesconsumer.producer;
+
+import org.apache.avro.Schema;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.common.ApplicationLogger;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.model.ChipsKafkaMessage;
+import uk.gov.companieshouse.kafka.message.Message;
+import uk.gov.companieshouse.kafka.producer.CHKafkaProducer;
+import uk.gov.companieshouse.service.ServiceException;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+@Component
+public class OutgoingMessageProducer implements MessageProducer {
```

Comment:
this bean is wired to use 
@Value("${kafka.retry.topic}")
    private String retryTopicName;

I would rename the class to indicate it is the producer for the retry topic, something like RetryMessageProducer

Reply:
I'd prefer to keep it as generic as possible as this won't be the only time a producer is required. I have put the field here to keep the code a simple as possible for now as we are oly addressing retry at the moment.

Reply:
in which case, just name it MessageProducerImpl, outgoing makes it sound like we're sending a message to another app and that there's a Incoming equivalent. If it is to be used for more than just retry, i would also rename the writeToQueue method to writeToRetryTopic.

Reply:
Name change for the class - the rest I've kep as is for now this will change when we add the error topic.